src/openvic-simulation/map/Map.cpp

```diff
@@ -69,6 +69,173 @@ bool Map::add_province(std::string_view identifier, colour_t colour) {
 	return provinces.add_item(std::move(new_province));
 }
 
+Province::distance_t Map::calculate_distance_between(Province const& from, Province const& to) {
+	const fvec2_t to_pos = to.get_unit_position();
+	const fvec2_t from_pos = from.get_unit_position();
+
+	const fixed_point_t min_x = std::min(
+		to_pos.x - from_pos.x,
+		std::min(
+			to_pos.x - from_pos.x + width,
+			to_pos.x - from_pos.x - width
```

Comment:
```suggestion
		(to_pos.x - from_pos.x).abs(),
		std::min(
			(to_pos.x - from_pos.x + width).abs(),
			(to_pos.x - from_pos.x - width).abs()
```
We need to take the absolute value otherwise it'll select large negative ones.

Reply:
oh, right!